Content.Tests/DMTests.cs

```diff
@@ -58,18 +59,29 @@ private static void Cleanup(string? compiledFile) {
     }
 
     [Test, TestCaseSource(nameof(GetTests))]
-    public void TestFiles(string sourceFile, DMTestFlags testFlags) {
+    public void TestFiles(string sourceFile, DMTestFlags testFlags, string errorCode) {
         string initialDirectory = Directory.GetCurrentDirectory();
         TestContext.WriteLine($"--- TEST {sourceFile} | Flags: {testFlags}");
         try {
-            string? compiledFile = Compile(Path.Join(initialDirectory, TestsDirectory, sourceFile));
             if (testFlags.HasFlag(DMTestFlags.CompileError)) {
-                Assert.That(compiledFile, Is.Null, "Expected an error during DM compilation");
-                Cleanup(compiledFile);
+                Assert.That(string.IsNullOrEmpty(errorCode), Is.False, "Expected an error code");
+                var originalConsoleOut = Console.Out;
+                StringWriter consoleOutput = new StringWriter();
+                Console.SetOut(consoleOutput);
+                string? compileErrorFile = Compile(Path.Join(initialDirectory, TestsDirectory, sourceFile));
+                string output = consoleOutput.ToString();
+                Console.SetOut(originalConsoleOut);
+                Console.WriteLine(output);
+                Assert.That(output.Contains(errorCode), Is.True, $"Expected error code \"{errorCode}\" was not found");
```

Comment:
Instead of checking the console output for `OD1234` could there instead be a HashSet on DMCompiler tracking which emission codes have been made? It could also be used to ensure no other codes have been raised either.